cookbooks/aws-parallelcluster-config/files/default/cloudwatch_agent/write_cloudwatch_agent_json.py

```diff
@@ -177,14 +178,68 @@ def filter_output_fields(configs):
     return [{desired_key: config[desired_key] for desired_key in desired_keys} for config in configs]
 
 
-def create_config(log_configs):
+def create_metrics_collected(selected_configs):
+    """Create the "metrics_collected" section in metrics configuration for selected metrics."""
+    metrics_collected = {}
```

Comment:
What about simplify the code to be something like:

(draft, to be verified)
```
def create_metrics_collected(selected_configs):
    “”"Create the “metrics_collected” section in metrics configuration for selected metrics.“”"
    desired_keys = [
        “measurement”,
        “metrics_collection_interval”,
        “resources”
    ]

    def _collect_metric(metric_config):
        collected = {key: metric_config.get(key) for key in metric_config if key in desired_keys}
        if “append_dimensions” in metric_config and “ClusterName” in metric_config[“append_dimensions”]:
            collected[“append_dimensions”] = {“ClusterName”: get_node_info().get(“stack_name”)}
        return collected

    return {metric_config[“metric_type”]: _collect_metric(metric_config) for metric_config in selected_configs}
```

Also, do we really need to append `ClusterName` dimension for now? It is not used in the cli to generate dashboard or create alarm?

Reply:
If we don't user `ClusterName` in the dashboard, what happens if a user creates two clusters in the same account? Will the dashboards pickup metrics from the other cluster or do we have another way of keeping the separate?

Reply:
Instance id or headnode private ip are used for metrics and alarm. We can use that to keep them separate

Reply:
code has been refactored as you suggested @chenwany